drivers/mdio/mdio_xilinx_axi_ethernet_lite.c

```diff
@@ -0,0 +1,233 @@
+		waited_cycles++;
+	}
+
+	if (waited_cycles == MDIO_MAX_WAIT_US) {
+		LOG_ERR("Timed out waiting for MDIO transaction to complete!");
+		return -EIO;
+	}
+	/* busy went low - transaction complete */
+	return 0;
+}
+
+static int xilinx_axi_ethernet_lite_mdio_read(const struct device *dev, uint8_t prtad,
+					      uint8_t regad, uint16_t *value)
+{
+	const struct mdio_xilinx_axi_ethernet_lite_config *config = dev->config;
+	struct mdio_xilinx_axi_ethernet_lite_data *data = dev->data;
+
+	if (prtad >= MDIO_XILINX_AXI_ETHERNET_LITE_MAX_PHY_DEVICES) {
+		LOG_ERR("Requested read port address %" PRIu8 " not supported - max %d", prtad,
+			MDIO_XILINX_AXI_ETHERNET_LITE_MAX_PHY_DEVICES);
+		return -ENOSYS;
+	}
+
+	(void)k_mutex_lock(&data->mutex, K_FOREVER);
+
+	if (!data->bus_enabled) {
+		LOG_ERR("MDIO bus not enabled!");
+		(void)k_mutex_unlock(&data->mutex);
+		return -ENOSYS;
+	}
+	if (mdio_xilinx_axi_ethernet_lite_check_busy(config)) {
+		LOG_ERR("MDIO bus busy!");
+		(void)k_mutex_unlock(&data->mutex);
+		return -ENOSYS;
+	}
+
+	mdio_xilinx_axi_ethernet_lite_set_addr(config, prtad, regad, true);
+
+	if (xilinx_axi_ethernet_lite_mdio_complete_transaction(config)) {
+		(void)k_mutex_unlock(&data->mutex);
+		return -EIO;
+	}
+
+	*value = (uint16_t)mdio_xilinx_axi_ethernet_lite_read_reg(
+		config, MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_READ_DATA_REG_OFFSET);
+
+	(void)k_mutex_unlock(&data->mutex);
+
+	return 0;
+}
+
+static int xilinx_axi_ethernet_lite_mdio_write(const struct device *dev, uint8_t prtad,
+					       uint8_t regad, uint16_t value)
+{
+	const struct mdio_xilinx_axi_ethernet_lite_config *config = dev->config;
+	struct mdio_xilinx_axi_ethernet_lite_data *data = dev->data;
+
+	if (prtad >= MDIO_XILINX_AXI_ETHERNET_LITE_MAX_PHY_DEVICES) {
+		LOG_ERR("Requested write port address %" PRIu8 " not supported - max %d", prtad,
+			MDIO_XILINX_AXI_ETHERNET_LITE_MAX_PHY_DEVICES);
+		return -ENOSYS;
+	}
+
+	(void)k_mutex_lock(&data->mutex, K_FOREVER);
+
+	if (!data->bus_enabled) {
+		LOG_ERR("MDIO bus not enabled!");
+		(void)k_mutex_unlock(&data->mutex);
+		return -ENOSYS;
+	}
+	if (mdio_xilinx_axi_ethernet_lite_check_busy(config)) {
+		LOG_ERR("MDIO bus busy!");
+		(void)k_mutex_unlock(&data->mutex);
+		return -ENOSYS;
+	}
+	mdio_xilinx_axi_ethernet_lite_set_addr(config, prtad, regad, false);
+	mdio_xilinx_axi_ethernet_lite_write_reg(
+		config, MDIO_XILINX_AXI_ETHERNET_LITE_MDIO_WRITE_DATA_REG_OFFSET, value);
+
+	if (xilinx_axi_ethernet_lite_mdio_complete_transaction(config)) {
+		(void)k_mutex_unlock(&data->mutex);
+		return -EIO;
+	}
+
+	(void)k_mutex_unlock(&data->mutex);
+
+	return 0;
+}
+
+static int xilinx_axi_ethernet_lite_mdio_init(const struct device *dev)
+{
+	struct mdio_xilinx_axi_ethernet_lite_data *data = dev->data;
+
+	(void)k_mutex_init(&data->mutex);
+
+	xilinx_axi_ethernet_lite_mdio_bus_enable(dev);
+
+	return 0;
+}
+
+
+static DEVICE_API(mdio, mdio_xilinx_axi_ethernet_lite_api) = {
+	/* clause 45 not supported */
```

Comment:
redundant comment, i can see that it is not implemented

Reply:
Done